### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by aquastripe** [zan8161/Plant_Seedlings_Classification#1]
1.
    ```python
    image_label = torch.zeros(self.outdim)
    image_label[img_label_idx] = 1
    ```
    參考 [nn.CrossEntropyLoss](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) 只要給 class index 就可以了，不需要自己做 one-hot encoding
2.
    ```python
    best_loss = sys.maxsize
    ```
    可以改成 `float('inf')` 會更合語意
3. `.item()` 可以直接 call 不用先 `.detach()`
4.
    ```python
    _, preds = torch.max(pred.data, 1)
    ```
    可以改用 `argmax` 會更簡潔，且 `.data` 是不需要的，這是舊版的 API
5. `TestDataSet` 的 `__getitem__` 可以放 filename 然後在 `test()` 裡面取出資料同時也取出檔名，這樣就不用處理兩次
6. 參數檔不要放 Git 比較好，通常會另外放在雲端硬碟供人下載
